### ISSUE TITLE
`npm start` is broken - `require.main` is undefined

### ISSUE THREAD

**Issue by AtkinsSJ** [HeyPuter/puter#467]
**Issue Description**
Error message:
```
❯ npm start

> puter.com@2.3.0 start
> node run-selfhosted.js

kerne [class Kernel extends AdvancedBase]
Boot logger started :)
[BOOT/ERROR] TypeError: Cannot read properties of undefined (reading 'filename')
    at RuntimeEnvironment.init_ (/home/sam/Projects/HeyPuter/puter/packages/backend/src/boot/RuntimeEnvironment.js:221:70)
    at RuntimeEnvironment.init (/home/sam/Projects/HeyPuter/puter/packages/backend/src/boot/RuntimeEnvironment.js:208:25)
    at Kernel._runtime_init (/home/sam/Projects/HeyPuter/puter/packages/backend/src/Kernel.js:59:40)
    at Kernel.boot (/home/sam/Projects/HeyPuter/puter/packages/backend/src/Kernel.js:67:14)
    at main (file:///home/sam/Projects/HeyPuter/puter/run-selfhosted.js:70:7)
    at async file:///home/sam/Projects/HeyPuter/puter/run-selfhosted.js:131:9

┏━━ [ HELP: `Cannot read properties of undefined (reading 'filename')` ]
┃ Whoops! Looks like something isn't working!
┃ No help is available for this error.
┃ Help can be added in src/errors/error_help_details.
┗━━ [ END HELP ]

```

That's this line:
https://github.com/HeyPuter/puter/blob/684b0dc059ba61c99d543749ed72e97cd405be1e/packages/backend/src/boot/RuntimeEnvironment.js#L221

This was added in 684b0dc059ba61c99d543749ed72e97cd405be1e. cc @KernelDeimos 

**Steps to Reproduce**
- Pull latest main
- `npm install` because the structure has changed
- `npm start`

**Expected & Actual Behavior**

**Addition Information or Screenshots (if applicable)**

**Deployment**
- [ ] Production (puter.com)
- [x] Development (`npm run start`)
- [ ] Docker (via `docker run`)
- [ ] Docker (via `docker-compose`)

**Puter version (if accessible)**
> Click the profile button on the top-right, then "Settings". The version number will be printed at the bottom

**Comment by enfermero-dev** [comment]
confirm this, freshly clone and can-t start. The parent commit works okay> 8d22276

**Comment by KernelDeimos** [comment]
This was my fault, this happened because the entrypoint is an ES6 module by _my_ entrypoint is not. Unfortunately despite the ESM-mechanism having the same value in a different place it doesn't get passed on when importing CJS modules (which seems odd given that it "supports" importing CJS modules)

Fixed by: 8e0f150fab7868ff69c7b314730cdec6407ac2aa
